models/core/blocks.sql

```diff
@@ -1,24 +1,37 @@
-{{ 
+{{
     config(
-        materialized='incremental', 
-        unique_key='block_id', 
+        materialized='incremental',
+        unique_key='block_id',
         tags=['core'],
         cluster_by=['block_timestamp']
-    ) 
+    )
 }}
 
-select 
-    block_id,
-    block_timestamp,
-    header:hash::string as block_hash,
-    header:parent_hash::string as block_parent_hash,
-    header:gas_limit as gas_limit,
-    header:gas_used as gas_used,
-    header:miner::string as miner,
-    header:nonce::string as nonce,
-    header:size as size,
-    tx_count,
-    header:state_root::string as state_root,
-    header:receipts_root::string as receipts_root
-from {{ deduped_blocks("harmony_blocks") }}
-where {{ incremental_load_filter("block_timestamp") }}
+with base_blocks as (
+
+    select * from {{ deduped_blocks("harmony_blocks") }}
+
+),
+
+final as (
+
+    select
+
+        block_id,
+        block_timestamp,
+        header:hash::string as block_hash,
+        header:parent_hash::string as block_parent_hash,
+        header:gas_limit as gas_limit,
+        header:gas_used as gas_used,
+        header:miner::string as miner,
+        header:nonce::string as nonce,
+        header:size as size,
+        tx_count,
+        header:state_root::string as state_root,
+        header:receipts_root::string as receipts_root
+
+    from base_blocks
+    where {{ incremental_load_filter("block_timestamp") }}
```

Comment:
I think the where clause can go into the CTE? Should be less load on the server..

Reply:
I did that based on [Tristan Handy's commentary here](https://discourse.getdbt.com/t/why-the-fishtown-sql-style-guide-uses-so-many-ctes/1091). Evidently, Snowflake treats the SELECT * as a pass-through, so there's actually no negative impact on performance.

Reply:
Welp, I got schooled :D